sdk/identity/azure-identity/azure/identity/aio/_credentials/managed_identity.py

```diff
@@ -68,7 +68,7 @@ def __init__(self, **kwargs: "Any") -> None:
 
             self._credential = TokenExchangeCredential(
                 tenant_id=os.environ[EnvironmentVariables.AZURE_TENANT_ID],
-                client_id=os.environ[EnvironmentVariables.AZURE_CLIENT_ID],
+                client_id=kwargs.pop("client_id", os.environ[EnvironmentVariables.AZURE_CLIENT_ID]),
```

Comment:
How is identity_config consumed?

Reply:
Currently it's ignored. The platform only accepts "client_id", so really there's no need for `identity_config`. Of course someone may use it anyway. There are some design considerations to think through before deciding how to handle it. We don't need a solution for this release, so I created #20575.

Reply:
Then should we remove 

```py
:keyword identity_config: a mapping ``{parameter_name: value}`` specifying a user-assigned identity by its object
        or resource ID, for example ``{"object_id": "..."}``. Check the documentation for your hosting environment to
        learn what values it expects.
:paramtype identity_config: Mapping[str, str]
```

from the docstring?

Reply:
No, all other implementations observe it.